### ISSUE TITLE
Setting editor control values through javascript

### ISSUE THREAD

**Issue by springfirenator** [pojome/elementor#4941]
We're a creating a custom functionality where we should set values to different controls in the editor panel. So far, we could change the underlying model like this:

```javascript
elementor.hooks.addAction( 'panel/open_editor/widget', function( panel, model, view ) {
    model.setSetting('text_field', 'sample text');
});
```

This works for basic controls like text box. For others, like choose control, the radio button doesn't visually switch to updated value. We need a way to update both view and model combined.

**Comment by bainternet** [comment]
@springfirenator 

You are correct, currently, its missing the `applySavedValue` which most other controls have, and that is why you are not seeing it updated in the panel.

**Comment by springfirenator** [comment]
Thanks, so is there any workaround for this? Perhaps calling applySavedValue in the hook works?

**Comment by AmibeWebsites** [comment]
Hi, I'd like to see this for select and select2 controls, being able to change the list of options. Thank you.

**Comment by Arkshine** [comment]
Any news on this or any known workaround? Just encountered the issue select2 options can't be updated. It's quite problematic.

**Comment by AmibeWebsites** [comment]
Not that I'm aware of. Only workaround I was able to come up with was to load multiple selects and conditionally hide/show them...

**Comment by dbjpanda** [comment]
Encountered the same as well. Wanted to change the button text. But no luck.

**Comment by dbjpanda** [comment]
@alexkuc Is it still working for you?

**Comment by alexkuc** [comment]
@dbjpanda At the time of posting, it was working for me. I haven't really touched Elementor-related code in a while. I don't really remember where `model.setSetting()` comes from otherwise I would post a code snippet.

**Comment by alexkuc** [comment]
@dbjpanda At the time of posting, it was working for me. I haven't really touched Elementor-related code in a while. I don't really remember where `model.setSetting()` comes from otherwise I would post a code snippet.
